meka/srcs/file.c

```diff
@@ -105,54 +104,72 @@ void    Filenames_Init()
         *p = EOSTR;
     else
         strcpy(g_env.Paths.EmulatorDirectory, g_env.Paths.StartingDirectory);
-
-    //ConsolePrintf ("g_env.Paths.StartingDirectory = %s\n", g_env.Paths.StartingDirectory);
-    //ConsolePrintf ("g_env.Paths.EmulatorDirectory = %s\n", g_env.Paths.EmulatorDirectory);
-    //ConsolePrintf ("argv[0] = %s\n", g_env.argv[0]);
+#endif
 
 #if defined(ARCH_UNIX) || defined(ARCH_MACOSX)
     {
 		char* rp = realpath(g_env.Paths.EmulatorDirectory, NULL);
 		if (rp != NULL)
 		{
 			strcpy(g_env.Paths.EmulatorDirectory, rp);
-			strcat(g_env.Paths.EmulatorDirectory, "/");
+			//strcat(g_env.Paths.EmulatorDirectory, "/");
 			free(rp);
 		}
     }
 #endif
 
+	char dataDirectory[FILENAME_LEN], internalResourcesDirectory[FILENAME_LEN], externalResourcesDirectory[FILENAME_LEN];
+	strcpy(dataDirectory, g_env.Paths.EmulatorDirectory);
+#ifdef ARCH_MACOSX
+	GetWritableInternalResourcePath( internalResourcesDirectory, sizeof(internalResourcesDirectory) );
+	GetWritableExternalResourcePath( externalResourcesDirectory, sizeof(externalResourcesDirectory) );
+	const char *filesToCopyToInternalResourcesPath[2] = {
+		"meka.inp",
+		NULL

```

Comment:
This patch actually causes an issue if you are compiling a basic binary and not a bundled .app folder (MEKA crashes).